### PR TITLE
Support fp16 in conversion from original ckpt

### PR DESCRIPTION
**Pull request by burgalon**

### CODE REVIEW

File: scripts/convert_original_stable_diffusion_to_diffusers.py [hunk 1]
```diff
@@ -148,3 +149,9 @@
         pipe.controlnet.save_pretrained(args.dump_path, safe_serialization=args.to_safetensors)
     else:
         pipe.save_pretrained(args.dump_path, safe_serialization=args.to_safetensors)
+
+    if args.half:
+        # convert pipe to half
+        pipe.unet = pipe.unet.half()
+        pipe.vae = pipe.vae.half()
+        pipe.text_encoder = pipe.text_encoder.half()
```
Comment (patrickvonplaten) [inline scripts/convert_original_stable_diffusion_to_diffusers.py:157]: This should probably be done before saving the pipeline no?

Comment (burgalon) [reply scripts/convert_original_stable_diffusion_to_diffusers.py]: my bad, thank you!

File: scripts/convert_original_stable_diffusion_to_diffusers.py [hunk 2]
```diff
@@ -143,8 +144,15 @@
         controlnet=args.controlnet,
     )
 
+    if args.half:
+        pipe.unet = pipe.unet.half()
```
Comment (patrickvonplaten) [inline scripts/convert_original_stable_diffusion_to_diffusers.py:148]:
```suggestion
        pipe.to(dtype=torch.float16)
```

File: scripts/convert_original_stable_diffusion_to_diffusers.py [hunk 3]
```diff
@@ -143,8 +144,15 @@
...
+        pipe.vae = pipe.vae.half()
```
Comment (patrickvonplaten) [inline scripts/convert_original_stable_diffusion_to_diffusers.py:149]:
```suggestion
```

**Comment by patrickvonplaten** [comment]
I think we can directly use the `.to(...)` method here - see docs: https://huggingface.co/docs/diffusers/main/en/api/diffusion_pipeline#diffusers.DiffusionPipeline.to it allows for `dtype` input now.

BTW @yiyixuxu we should probably add some nice docstring there

File: scripts/convert_original_stable_diffusion_to_diffusers.py (hunk 2 above)
Comment (burgalon) [reply scripts/convert_original_stable_diffusion_to_diffusers.py:148]: ah! i didn't know that's possible! Thank you

**Comment by burgalon** [comment]
> I think we can directly use the `.to(...)` method here

done

**Comment by patrickvonplaten** [comment]
Thanks for the fixes!

**Review by patrickvonplaten — APPROVED** [review]

**Pull request merged**